src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/PrimitiveArrayMutatorFactory.java

```diff
@@ -145,38 +149,66 @@ private void extractRange(AnnotatedType type) {
 
       switch (type.getType().getTypeName()) {
         case "int":
-          minRange = inRange.map(InRange::min).orElse((long) Integer.MIN_VALUE);
-          maxRange = inRange.map(InRange::max).orElse((long) Integer.MAX_VALUE);
+          {
+            LongRange r =
+                RangeSupport.resolveIntegralRange(
+                    type, (long) Integer.MIN_VALUE, (long) Integer.MAX_VALUE);
+            minRange = r.min;
+            maxRange = r.max;
+          }
           break;
         case "long":
-          minRange = inRange.map(InRange::min).orElse(Long.MIN_VALUE);
-          maxRange = inRange.map(InRange::max).orElse(Long.MAX_VALUE);
+          {
+            LongRange r = RangeSupport.resolveIntegralRange(type, Long.MIN_VALUE, Long.MAX_VALUE);
+            minRange = r.min;
+            maxRange = r.max;
+          }
           break;
         case "short":
-          minRange = inRange.map(InRange::min).orElse((long) Short.MIN_VALUE);
-          maxRange = inRange.map(InRange::max).orElse((long) Short.MAX_VALUE);
+          {
+            LongRange r =
+                RangeSupport.resolveIntegralRange(
+                    type, (long) Short.MIN_VALUE, (long) Short.MAX_VALUE);
+            minRange = r.min;
+            maxRange = r.max;
+          }
           break;
         case "char":
           minRange = inRange.map(InRange::min).orElse((long) Character.MIN_VALUE);
```

Comment:
I think, we should be able to set  `@InRange` for `Character` and `char` types (they are currently not in the list of `@AppliesTo`), so the old code always used Character.MIN/MAX

Reply:
Do you think I should add this change in this PR or create a follow-up? I'm not sure how big this change would be?